crates/storage/provider/src/providers/static_file/manager.rs

```diff
@@ -143,9 +143,9 @@ impl StaticFileProvider {
                         // appending/truncating rows
                         for segment in event.paths {
                             // Ensure it's a file with the .conf extension
-                            if !segment
+                            if segment
                                 .extension()
-                                .is_some_and(|s| s.to_str() == Some(CONFIG_FILE_EXTENSION))
+                                .is_none_or(|s| s.to_str() != Some(CONFIG_FILE_EXTENSION))
```

Comment:
This is more readable with the current is_some_and

Reply:
yeah kind of agreed but this is clippy new suggestion... not sure this is good to put an allow() on this isolate case but if you have a strong opinion there I can revert :)

Reply:
yes please allow it

Reply:
done